SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/kyc/view/IdentityVerificationViewModel.kt

```diff
@@ -0,0 +1,271 @@
+                    }
+                    waitFor.await()
+                }
+            }
+        }
+    }
+
+    fun getCustomerStatus() {
+
+        Cybrid.instance.let { cybrid ->
+            if (!cybrid.invalidToken) {
+                viewModelScope.let { scope ->
+                    scope.launch {
+
+                        val customerResult = getResult {
+                            customerService.getCustomer(
+                                customerGuid = customerGuid)
+                        }
+                        Log.d("DXGOP", "CUSTOMER STATUS")
+                        Log.d("DXGOP", customerResult.toString())
+                        checkCustomerStatus(customerResult.data?.state ?: CustomerBankModel.State.storing)
+                    }
+                }
+            }
+        }
+    }
+
+    fun getIdentityVerificationStatus(record: IdentityVerificationBankModel? = null) {
+
+        Cybrid.instance.let { cybrid ->
+            if (!cybrid.invalidToken) {
+                viewModelScope.let { scope ->
+                    scope.launch {
+
+                        var lastVerification = record ?: getLastIdentityVerification()
+                        Log.d("DXGOP", "---> LAST VERIFICATION")
+                        Log.d("DXGOP", lastVerification.toString())
+
+                        if (lastVerification == null ||
+                            lastVerification.state == IdentityVerificationBankModel.State.expired ||
+                            lastVerification.personaState == IdentityVerificationBankModel.PersonaState.expired) {
+
+                            lastVerification = createIdentityVerification()
+                            Log.d("DXGOP", "---> CREATION")
+                            Log.d("DXGOP", lastVerification.toString())
+                        }
+                        val recordResponse = getResult {
+                            identityService.getIdentityVerification(
+                                identityVerificationGuid = lastVerification?.guid ?: ""
+                            )
+                        }
+                        Log.d("DXGOP", "IDENTITY STATUS")
+                        Log.d("DXGOP", recordResponse.toString())
+                        checkIdentityRecordStatus(recordResponse.data)
+                    }
+                }
+            }
+        }
+    }
+
+    private suspend fun getLastIdentityVerification(): IdentityVerificationBankModel? {
+
+        var verification: IdentityVerificationBankModel? = null
+        Cybrid.instance.let { cybrid ->
+            if (!cybrid.invalidToken) {
+                viewModelScope.let { scope ->
+                    val waitFor = scope.async {
+
+                        val identityResponse = getResult {
+                            identityService.listIdentityVerifications(
+                                customerGuid = customerGuid,
+                                page = JavaBigDecimal(1),
```

Comment:
Believe the page would be `0`, not `1`; page numbers are zero-indexed.

Reply:
Done!